uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -246,4 +246,42 @@ public static function getMostRecentStartDateAvailable()
 
     return date('Y-m-d');
   }
+
+  /**
+   * Returns the number of working days for this Absence Period.
+   *
+   * The number is given by "number of ways in period" - "weekends" - "public holidays".
+   *
+   * If a public holiday falls on a weekend, we only count one day.
+   *
+   * @return int
+   */
+  public function getNumberOfWorkingDays()
+  {
+    $startDate = new DateTime($this->start_date);
+    $endDate = new DateTime($this->end_date);
+    $oneDayInterval = new DateInterval('P1D');
+
+    // DatePeriod doesn't include the end date,
+    // so we add one more day for it to be included
+    $endDate->add($oneDayInterval);
+
+    $numberOfWorkingDays = 0;
+    $period = new DatePeriod($startDate, $oneDayInterval, $endDate);
+    foreach($period as $date) {
+      $dayOfTheWeek = $date->format('N');
+      $dayIsWorkingDay = $dayOfTheWeek > 0 && $dayOfTheWeek < 6;
```

Comment:
About the flexible weekends, we don't need to consider them for this calculation.
Now, about the constants, I don't think they will improve the code on this specific situation. Here's why:
- I would need to add the constants to the class even though they have nothing to do with an AbsencePeriod. Something like AbsencePeriod::SATURDAY doesn't make any sense.
- I would only use the constants in a single method. That is a code smell indicating it should be extracted.
- Class constants in PHP are always public (this is change in PHP 7.1), so It means we are exposing some not very good code to be used elsewhere.

I thought about all this while developing and that's why I decided to include the $daysIsWorkingDay variable. Here my reasoning:
A working day is pretty much the same everywhere in the world. Except for a very specific case (that I have no knowledge of) working days are always monday to friday. So, when I say "a day is a working day if its day of the week is greater than 0 and less than 6" it's kind of implicit that each days of the week is a number starting with from Sunday (0) and ending in Saturday (6).

I know, it's not perfect but I think it's better than the constants.